### PR TITLE
TYP: annotate plotting._matplotlib.converter

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/plotting/_matplotlib/converter.py [hunk 1]
```diff
@@ -473,7 +473,7 @@ def _get_default_annual_spacing(nyears) -> Tuple[int, int]:
     return (min_spacing, maj_spacing)
 
 
-def period_break(dates, period):
+def period_break(dates: PeriodIndex, period: str) -> np.ndarray:
```
Comment (simonjayhawkins) [inline pandas/plotting/_matplotlib/converter.py:476]: the docsting says that dates is "dates : PeriodIndex" and goes on to say "Array of intervals to monitor." could a PeriodArray be passed?

**Review by simonjayhawkins — APPROVED** [review]
Thanks @jbrockmendel lgtm pending green

File: pandas/plotting/_matplotlib/converter.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/plotting/_matplotlib/converter.py:476]: It is only ever called with PeriodIndex.  I guess passing PeriodArray wouldnt break the world.

Comment (simonjayhawkins) [reply pandas/plotting/_matplotlib/converter.py:476]: while period_range returns a PeriodIndex there doesn't look like anything to gain by adding PeriodArray atm

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**